ndcube/tests/helpers.py

```diff
@@ -136,8 +136,14 @@ def assert_cubes_equal(test_input, expected_cube, check_data=True, check_uncerta
             assert type(test_input.uncertainty) is type(expected_cube.uncertainty)
             assert np.allclose(test_input.uncertainty.array, expected_cube.uncertainty.array), \
                 f"Expected uncertainty: {expected_cube.uncertainty}, but got: {test_input.uncertainty.array}"
+        else:
+            if test_input.uncertainty is None:
+                assert expected_cube.uncertainty is None, "Expected cube's uncertainty should also be None"
+            elif expected_cube.uncertainty is None:
+                assert test_input.uncertainty is None, "test_input should also be None"
             else:
-                assert (test_input.uncertainty is None and expected_cube.uncertainty is None)
+                pass
```

Comment:
Now the way you have it, you could do this:
```suggestion
                assert type(test_input.uncertainty) is type(expected_cube.uncertainty)
                assert np.allclose(test_input.uncertainty.array, expected_cube.uncertainty.array), \
                f"Expected uncertainty: {expected_cube.uncertainty}, but got: {test_input.uncertainty.array}"
```
and then remove lines 135-139.  Up to you though.

Reply:
@DanRyanIrish Thank you for your suggestion. I re-understood what the code aims to do, and changed it again.